### DISCUSSION TITLE
Matrices

### DISCUSSION THREAD

**Discussion by mmasias** [mmasias/24-25-PRG1#641] [Announcements]
Luego de la clase de hoy, quedan en el repo:

1. La extensión de la sección [**¿Cómo?**](https://github.com/mmasias/PRG1/blob/main/temario/00800-matrices.md#c%C3%B3mo---en-java), incluyendo las profundas reflexiones en la que le damos vuelta al concepto de que "igual" no es lo mismo que "lo mismo"
1. La [hoja de ruta](https://github.com/mmasias/mmasias/blob/main/UNEATLANTICO/prg1/referencias.md) que seguimos en clase.
1. [El código fuente](https://github.com/mmasias/24-25-PRG1/blob/main/entregas/masiasManuel/src/Matrices.java) que aplica lo que vimos.

Queda este hilo abierto a debate, discusiones y mejoras.

Manuel

**Comment by JavierLianoRioz** [comment]
Dices en el repo:

> Es altamente recomendado que la matriz se trate de una colección de elementos homogéneos, todos ellos del mismo tipo y de la misma naturaleza: no combinar 5 contadores y un acumulador

Entiendo por esto que se podría tener un array que tenga múltiples arrays (una de 2D) que nos aporte diferente información por array. Pero no sabría como definir este array.

Ejemplo:
```
String[] nombresDeComida = {"pollo", "carne", ...}
int[] cantidadDeComida = {5, 7, ...}
???[] Frigorifico = {nombreDeComida, cantidadDeComida, ...}
```
Para asemejar a la tabla
| Nombre | Cantidad |
|---|---|
| Pollo | 5 |
| Carne | 7 |
|...|...|

**Reply by mmasias** [reply]
¿Cómo no he visto esto antes? Entono el _mea culpa_ y procedo a contestar... Cuando hablamos de elementos homogéneos lo hacemos dentro de la naturaleza y limitaciones de una matriz. Y esto es el tipo de datos: [por definición](https://github.com/mmasias/PRG1/blob/main/temario/00800-matrices.md#qu%C3%A9) una matriz es una colección de elementos, todos del mismo tipo. La tabla que propones tiene datos de tipo String y datos de tipo entero, lo que hace impracticable el resolverlo con una única matriz (salvo que almacenes la cantidad como String, pero eso es forzado). Para resolver esta limitación se escaló hacia los tipos de datos compuestos, que fueron una de las semillas de las clases. Los veremos!!!
